Add local automated notebook tests

Arun-Niranjan opened the pull request:
This change addresses https://github.com/infer-actively/pymdp/issues/293 - adding the infrastructure required for automated testing of notebooks. this includes:

- the addition of a test dependency group that has the existing test dependencies like pytest. [nbval](https://nbval.readthedocs.io/en/latest/) is the main new dependency for notebook evaluation.
⚠️ pytest has been moved out of the core dependency group. This means that to be able to run tests you *must* run `uv sync --group test` to get the correct dependencies.
- adds a readme explaining how to run the notebook tests locally with nbval

eventually with a bit more tidy-up, we can run these tests in CI (either on every PR or a nightly build if they take too long)

this PR is largely reproduced from https://github.com/infer-actively/pymdp/pull/294 to account for the merge from chore/notebook to v1 alpha branch

conorheins commented on test/notebooks/test_notebooks.py:
```diff
@@ -0,0 +1,134 @@
+
+@pytest.mark.notebook
+@pytest.mark.slow
+@pytest.mark.skip("model fitting notebook is currently being migrated, see https://github.com/infer-actively/pymdp/issues/287")
```
I think this pytest.mark.skip can be removed now, because the notebook is currently running (but the pytest.mark.slow is certainly warranted)

conorheins commented on pyproject.toml:
```diff
@@ -32,7 +32,16 @@ dependencies = [
+    "ipykernel>=7.1.0",
+    "jupyter>=1.1.1",
+    "nbval>=0.11.0",
+    "pybefit",
```
in the same way you're adding pybefit as a dependency to the `test` dependencty group, to be able to run the model-fitting notebook, can you also add the `mediapy` and `pygraphviz` dependencies here? otherwise people won't be able to run the notebook tests without all cells failing on tmaze_demo.ipynb, which requires mediapy (and is a pretty foundational notebook among the new jax suite)

conorheins commented on .github/workflows/test.yaml:
```diff
@@ -20,9 +20,13 @@ jobs:
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y graphviz-dev
```
thanks for catching that -- also for people running locally with the `test` dep group we're gonna have to deal with this somehow: https://github.com/infer-actively/pymdp/issues/268, but let's leave that to a future PR

Arun-Niranjan commented:
> Looks great thanks for addressing all those comments -- one final request before we merge - can you just squash the commits together into one ? I'm trying to keep the commit # as small as possible in advance of the v1.0.0_alpha --> master PR

yep there's a "squash and merge" button in github which does this for us 😄
